src/foam/core/referral/IdReferralCodeGenerationService.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.core.referral',
+  name: 'IdReferralCodeGenerationService',
+  documentation: 'Default referral code generation service that uses referral code ID as the code',
+
+  implements: [
+    'foam.core.referral.ReferralCodeGeneratorService'
+  ],
+
+  javaImports: [
+    'foam.lang.X',
+    'foam.lang.XLocator',
+    'foam.core.auth.User',
+    'foam.dao.DAO',
+    'foam.mlang.sink.Count',
+    'static foam.mlang.MLang.*'
+  ],
+
+  methods: [
+    {
+      name: 'getCode',
+      synchronized: true,
+      javaCode: `
+        X x = XLocator.get();
+        DAO referralCodeDAO = (DAO) x.get("referralCodeDAO");
```

Comment:
Not sure if this saves us any performance in java but if you also pass the referral code that you already have when calling this method, it could just return the referralCode.getId() rather than doing another dao call.

Reply:
Makes things a little messier as we would have to name each method differently, as one doesn't always have the previous code. 
Also, we'd have to validate the passed referral code and user id. 
MDAOs are fast, for future if needed.